src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -0,0 +1,603 @@
+            auto d1_    = sc.get_mem<cuDataType *>(ih, d1_acc);
+            auto d2_    = sc.get_mem<cuDataType *>(ih, d2_acc);
+            auto x1_    = sc.get_mem<cuDataType *>(ih, x1_acc);
+            auto y1_    = sc.get_mem<cuDataType *>(ih, y1_acc);
+            auto param_ = sc.get_mem<cuDataType *>(ih, param_acc);
+            cublasStatus_t err;
+            CUBLAS_ERROR_FUNC(func, err, handle, d1_, d2_, x1_, y1_, param_);
+        });
+    });
+}
+
+#define ROTMG_LAUNCHER(TYPE, CUBLAS_ROUTINE)                                          \
+    void rotmg(cl::sycl::queue &queue, cl::sycl::buffer<TYPE, 1> &d1,                 \
+               cl::sycl::buffer<TYPE, 1> &d2, cl::sycl::buffer<TYPE, 1> &x1, TYPE y1, \
+               cl::sycl::buffer<TYPE, 1> &param) {                                    \
+        rotmg(CUBLAS_ROUTINE, queue, d1, d2, x1, y1, param);                          \
+    }
+
+ROTMG_LAUNCHER(float, cublasSrotmg)
+ROTMG_LAUNCHER(double, cublasDrotmg)
+#undef ROTMG_LAUNCHER
+
+template <typename Func, typename T>
+inline void iamax(Func func, cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<T, 1> &x,
+                  const int64_t incx, cl::sycl::buffer<int64_t, 1> &result) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    overflow_check(n, incx);
+    // cuBLAS does not support int64_t as return type for the data. So we need to
+    // mimic iamax we are converting the result to be the int and then we convert
+    // it back to the actual data on the host.
+    // This change may cause failure as the result of integer overflow
+    // based on the size. Alternatively either we need to write two a sycl kernel
+    // to elementwise copy the data between two buffer, or allow reinterpret cast
+    // to convert to different type with different typesize size.
+    cl::sycl::buffer<int, 1> int_res_buff{ cl::sycl::range<1>(1) };
+    queue.submit([&](cl::sycl::handler &cgh) {
+        auto x_acc       = x.template get_access<cl::sycl::access::mode::read>(cgh);
+        auto int_res_acc = int_res_buff.template get_access<cl::sycl::access::mode::write>(cgh);
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc     = CublasScopedContextHandler(queue);
+            auto handle = sc.get_handle(queue);
+            // By default the pointer mode is the CUBLAS_POINTER_MODE_HOST
+            // when the data is on buffer, it must be set to
+            // CUBLAS_POINTER_MODE_DEVICE mode otherwise it causes the segmentation
+            // fault. When it is set to device it is users responsibility to
+            // synchronise as the function is completely asynchronous.
+            cublasSetPointerMode(handle, CUBLAS_POINTER_MODE_DEVICE);
+            auto x_       = sc.get_mem<cuDataType *>(ih, x_acc);
+            auto int_res_ = sc.get_mem<int *>(ih, int_res_acc);
+            cublasStatus_t err;
+            // IAMAX does not support negative incx
```

Comment:
if that is the case, why incx is passed without std::abs()?

Reply:
I think the wording did not convey the meaning. For `iamax` when the `incx` is negative cuBLAS returns 0 and does not execute the Kernel. This behavior is similar to that of reference NetlibBLAS. Because of that, we don't see the difference between the reference and cuBLAS. I think Intel's implementation is the same. So the `incx` should not have abs around it because by adding abs, it convert the `incx` to positive and the result would be different.